bundles/sirix-core/src/main/java/org/sirix/io/combined/CombinedStorageWriter.java

```diff
@@ -29,29 +31,61 @@ public CombinedStorageWriter(Writer localWriter, Writer remoteWriter, Reader sto
 	@Override
 	public Writer write(PageReadOnlyTrx pageReadOnlyTrx, PageReference pageReference, Bytes<ByteBuffer> bufferedBytes) {
 		Writer writer = localStorageWriter.write(pageReadOnlyTrx, pageReference, bufferedBytes);
-		remoteStorageWriter.write(pageReadOnlyTrx, pageReference, bufferedBytes);
+		CompletableFuture<Writer> remoteWriterTask = CompletableFuture.supplyAsync(() -> remoteStorageWriter.write(pageReadOnlyTrx, pageReference, bufferedBytes));
+		if (writer == null) {
```

Comment:
I think this branch is unnecessary, as the local writer will always return itself.

Reply:
Yes I agree to that, but wasn't completely sure about it. Will remove it in sometime - same for below comments

Reply:
Done.